src/SyntaxHighlighter/X86Assembly.cpp

```diff
@@ -216,9 +219,35 @@ const QRegularExpression kPlatformRegex{"^Platform:.*$"};
 X86Assembly::X86Assembly() : QSyntaxHighlighter{static_cast<QObject*>(nullptr)} {}
 
 void X86Assembly::highlightBlock(const QString& code) {
-  HighlightBlockAssembly(code, [this](int start, int count, const QTextCharFormat& format) {
-    setFormat(start, count, format);
-  });
+  const auto* const highlighting_metadata =
+      static_cast<const HighlightingMetadata*>(currentBlock().userData());
+  if (highlighting_metadata->IsMainContentLine()) {
```

Comment:
`highlighting_metadata` might be `nullptr`. We have to check that. If it's `nullptr` we can assume that it's main content.

Reply:
yes, yes. I was making that fixup, you were faster than me :P

Reply:
BTW, it's there